types/office-js/index.d.ts

```diff
@@ -551,6 +551,37 @@ declare namespace Office {
         */
         asyncContext: any;
         /**
+        * Gets an object that may provide additional information if an error occurred.
```

Comment:
Consider: "an object that may provide additional information if an error occurred" -> "an object with additional information about the error (if one occurred)"

Reply:
I think that what I currently have captures the gist of it. The diagnostics object does not provide info in every error case as it's only available for those two APIs.